_posts/2018-09-26-artsy-frontend-history.markdown

```diff
@@ -0,0 +1,172 @@
+of those projects.
+
+## Ezel
+
+In November 2013, we split our web frontend from the API backend. You can read
+[all the details in this blog post](2013_review), but the story is summarized nicely as "moving from a single
+monolithic application to modular Backbone apps that run in Node and the browser and consume our external API."
+This move from monolith to modular systems continues to influence day-to-day work on the Artsy Engineering team.
+
+We moved our API from Rails to [Grape][], partially motivated by a desire to build an iOS application that would
+consume this API. We faced a lot of problems with the split, including SEO problems, severe page load times,
+maintaining duplicated backend and frontend UI templates, slow test suites, and poor developer productivity. We
+took the project of building our mobile web frontend, m.artsy.net (still known as "martsy" internally) as an
+opportunity to address these problems.
+
+We built our new site with [Node.js][node] since it easily allowed server-side rendering. We split out areas of
+concern into separate "apps", with their own bundled CSS/JS to help page load times. We server-side rendered
+above-the-fold content and used client-side JS to load the rest, which helped SEO and user experience. We took a
+[BEM][]-like approach to our CSS, which helped developer productivity. Our technical decisions were driven
+primarily by our desire to create great user experiences.
+
+And because we are an open source by default organization, we collected these approaches into an open source
+project called [Ezel][].
+
+We used Ezel for a few years without too much change in our web frontend stack. It proved really useful for helping
+build other web apps – CMS systems for our partners, auction-management systems for our admins, all kinds of
+projects – and most of those projects started on Heroku before moving to heavier-duty deployments as needed. Our
+frontend mindset at the time (2015) was focused on getting to a stable, predictable stack. However... we started
+experimenting with React around the same time.
+
+CoffeeScript and Backbone were still working for us, and we still use them in production in many systems. However,
+the state of the art in web development moved on. When I joined the auctions team and helped maintain one of our
+CoffeeScript+Backbone apps, I was _very_ confused about how data flowed from one part of the app to another, across
+languages, with a lot of magic happening. I think that's typical in these kinds of apps – "convention over
+configuration" is a good mantra _if_ you can expect that incoming engineers are familiar with the conventions.
+That's just not the case anymore.
+
+By 2016, we launched our [first app built with React][auctions], which both proved the technology was ready for
+production use _and_ convinced our engineers that React is simply a better paradigm for building the kinds of user
```

Comment:
I think prediction was our first big react/redux stack, but I recall that other projects and teams were using react at this time too. Maybe we can tone down this paragraph a little to talk about how seminal prediction was because it went all-in on react or was a good expression of react because of the nature of the auctions product?

Reply:
Gotcha, so Prediction was really our first _from scratch_ React app, then?